draft-irtf-cfrg-hpke.md

```diff
@@ -1207,6 +1207,34 @@ across all labels in this document.
 The `0xFFFF` AEAD ID is reserved for applications which only use the Export
 interface; see {{hpke-export}} for more details.
 
+# API Considerations {#api-considerations}
+
+The high-level HPKE APIs specified in this document are all fallible.
+For example, `Decap()` can fail if the encapsulated key `enc` is invalid,
+and `Open()` may fail if ciphertext decryption fails. The explicit errors
+generated throughout this specification, along with the conditions that
+lead to each error, are as follows:
+
+- `ValidationError`: Public key validation failure; {{dhkem}}.
+- `DeserializeError`: Public or private key deserialization failure; {{base-crypto}}.
+- `DecapError`: `Decap()` failure; {{base-crypto}}.
+- `OpenError`: Context AEAD `Open()` failure; {{base-crypto}} and {{hpke-dem}}.
+- `MessageLimitReachedError`: Context AEAD sequence number overflow; {{base-crypto}} and {{hpke-dem}}.
+
+Implicit errors may also occur. As an example, certain classes of failures,
```

Comment:
I would emphasize that this is not an exhaustive list.  For example, an implementation might simply run out of memory.